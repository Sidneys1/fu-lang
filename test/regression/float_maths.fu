main: i32() = {
  a: f32 = 50f;
  b: f32 = 4f;
  c: f32 = a - b; // 46f
  d: f32 = a * b; // 200f
  e: f32 = a / b; // 12.5f
  return c + d + e; // 46 + 200 + 12.5 = 258.5 (258 as int)
};
