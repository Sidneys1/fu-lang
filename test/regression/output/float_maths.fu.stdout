---------- test\regression\float_maths.fu ----------
[2m   4 |   c: f32 = [0;2ma - b;[2m // 46f[0m[0;0m <-- Checks for infix operator '-' are not implemented! [0;2m(test\regression\float_maths.fu:4:12-17)
[0m[m[0;2m-----+[0m
[2m   5 |   d: f32 = [0;2ma * b;[2m // 200f[0m[0;0m <-- Checks for infix operator '*' are not implemented! [0;2m(test\regression\float_maths.fu:5:12-17)
[0m[m[0;2m-----+[0m
[2m   6 |   e: f32 = [0;2ma / b;[2m // 12.5f[0m[0;0m <-- Checks for infix operator '/' are not implemented! [0;2m(test\regression\float_maths.fu:6:12-17)
[0m[m[0;2m-----+[0m
[2m   7 |   return [0;2mc + d [2m+ e; // 46 + 200 + 12.5 = 258.5 (258 as int)[0m[0;0m <-- Checks for infix operator '+' are not implemented! [0;2m(test\regression\float_maths.fu:7:10-15)
[0m[m[0;2m-----+[0m
[2m   7 |   return [0;2mc + d + e;[2m // 46 + 200 + 12.5 = 258.5 (258 as int)[0m[0;0m <-- Checks for infix operator '+' are not implemented! [0;2m(test\regression\float_maths.fu:7:10-19)
[0m[m[0;2m-----+[0m
[2m   6 |   e: f32 = a / b; // 12.5f[0m
[2m   7 |   return [0;1mc + d + e;[2m // 46 + 200 + 12.5 = 258.5 (258 as int)[0m
[2m     |[;33m          ^^^^^^^^^^ Warning: Loss of precision converting from a `f32` to a `i32`.[0;2m (test\regression\float_maths.fu:7:10-19)
[0;2m   8 | }; [0m
[m